audb/core/load.py

```diff
@@ -163,6 +163,27 @@ def _database_is_complete(
     return complete
 
 
+def _files_duration(
+        db: audformat.Database,
+        deps: Dependencies,
+        files: typing.Sequence[str],
+        format: typing.Optional[str],
+):
+
+    def fix_file(file: str):
+        file = os.path.join(db.root, file)
+        if format is not None:
+            file = audeer.replace_file_extension(file, format)
+        return file
+
+    field = define.DEPEND_FIELD_NAMES[define.DependField.DURATION]
+    durs = deps._df.loc[files][field]
+    durs = durs[durs > 0]
+    durs = pd.to_timedelta(durs, unit='s')
```

Comment:
Turns out that this line takes the most of the time, e.g. 1.6 s from 2.4 s overall when running on `librispeech`.
The question is if it would make sense to move it to a later stage and only call it inside `Database.files_duration()`. On the other hand the idea behind `db._files_duration` is that we cache the result there and it should return very fast if it contains the entry. This would then no longer be tru as we would need to convert to `timedelta` first.

Reply:
The next big contributor is then `fix_file` which takes 0.5 s.

Reply:
We could store `pd.Timedelta` directly in the dependency table and only convert it to seconds when we write and read to CSV (for compatibility on Artifactory). In the pickled version in the cache we can again save `pd.Timedelta`.

Reply:
> The next big contributor is then `fix_file` which takes 0.5 s.

It might be faster to directly manipulate `index.levels` than calling `map()`.

Reply:
Good idea, the only problem is that you need to clean the cache in between to not break it with the old dependency files. Or maybe we can add a check and solve it with an `if`-statement. Otherwise it sounds like a nice solution.

For `fix_file()` I'm also wondering if maybe we don't have to call it every time as we also have `_full_path()` inside load. Maybe it would be sufficient to just call one of them?

Reply:
> For `fix_file()` I'm also wondering if maybe we don't have to call it every time as we also have `_full_path()` inside load. Maybe it would be sufficient to just call one of them?

The problems is that in the dependency table we have relative file names. So if we start from full files, we need to convert it then to relative files names to access the entries in the dependency table.

But: what we could do is re-use the full files in `db._files_duration` when we extend the paths in the tables later on. Though, we may not have an entry for every file due to 0 durations...

Reply:
I created https://github.com/audeering/audb/issues/144 to handle caching the timedelta values directly.

Reply:
I just realized that we currently fix the media extension (if required) and conversion to full path in two separate steps: `_fix_media_ext` and `_full_path`, this could also be combined. So I think what we need is maybe a pre-step where we calculate the file names we want to return with the database only once and then reuse it when we set `db._files_duration` and fixing the path in the tables.

Reply:
OK, but it also sounds like we should not combine it with this pull request.
I created https://github.com/audeering/audb/issues/145 to track this.